src/Raven.Studio/typescript/viewmodels/database/indexes/deleteIndexesConfirm.ts

```diff
@@ -3,35 +3,85 @@ import dialog = require("plugins/dialog");
+        this.indexName = name;
+        this.reduceOutputCollection = reduceCollection;
+        this.ReferenceCollection = hasReferenceCollection ? this.reduceOutputCollection + this.referenceCollectionExtension : "";
+    }
+}
 
-    constructor(private indexNames: string[], private db: database, title?: string) {
+class deleteIndexesConfirm extends dialogViewModelBase {
+    title: string;
+    subTitleHtml: string;
+    warning: KnockoutComputed<string>;
+    deleteTask = $.Deferred<boolean>();
+    
+    indexesInfoForDelete = ko.observableArray<indexInfoForDelete>();
+    
+    constructor(private indexes: index[] | indexDefinition[], private db: database) {
         super();
+        this.initObservables();
 
-        if (!indexNames || indexNames.length === 0) {
+        if (!indexes || indexes.length === 0) {
             throw new Error("Indexes must not be null or empty.");
         }
-
-        this.title = title || (indexNames.length === 1 ? "Delete index?" : "Delete indexes?");
+        
+        if (indexes[0] instanceof index) {
+            // Coming here from Indexes List View
+            this.indexesInfoForDelete((indexes as index[]).map(x => {
+                return new indexInfoForDelete(x.name, x.reduceOutputCollection(), !!x.reduceOutputReferencePattern()); 
+            }));
+        } else {
+            // Coming here from Edit-Index View
+            this.indexesInfoForDelete((indexes as indexDefinition[]).map(x => {
+                return new indexInfoForDelete(x.name(), x.reduceToCollectionName(), x.patternForOutput());
+            }));
+        }
+        
+        if (this.indexesInfoForDelete().length === 1) {
+            this.title = "Delete index?";
+            this.subTitleHtml = `You're deleting index:`;
+        } else {
+            this.title = "Delete indexes?";
+            this.subTitleHtml = `You're deleting <strong>${this.indexesInfoForDelete().length}</strong> indexes:`;
+        }
     }
 
     deleteIndexes() {
-        const deleteTasks = this.indexNames.map(name => new deleteIndexCommand(name, this.db).execute());
+        const deleteTasks = this.indexesInfoForDelete().map(indexItem => new deleteIndexCommand(indexItem.indexName, this.db).execute());
 
         $.when.apply($, deleteTasks)
             .done(() => {
-                if (this.indexNames.length > 1) {
-                    messagePublisher.reportSuccess("Successfully deleted " + this.indexNames.length + " indexes!");
+                if (this.indexesInfoForDelete.length > 1) {
+                    messagePublisher.reportSuccess("Successfully deleted " + this.indexesInfoForDelete.length + " indexes!");
                 }
                 this.deleteTask.resolve(true);
             })
             .fail(() => this.deleteTask.reject());
+
         dialog.close(this);
     }
+    
+    initObservables() {
+        this.warning = ko.pureComputed<string>(() => {            
+            const warningText =  `Note: <br>  
```

Comment:
can we move html to html file and expose boolean variable here and in html use: data-bind="visible: thisVariableName" ?

Reply:
yes - done